### PR TITLE
Bump version to 2.5.0 and document release process.

### PR DESCRIPTION
**Pull request by ScottTodd**
The `docs/releasing.md` file was first introduced next to the `build_release.py` script in https://github.com/nod-ai/SHARK-ModelDev/pull/605. I don't see that file in this repository (iree-org/iree-turbine) though, despite https://github.com/nod-ai/SHARK-ModelDev/pull/655 claiming that it had been upstreamed. Since the packages involved have changed over the months (shark-turbine --> iree-turbine naming, see also https://github.com/iree-org/iree-turbine/issues/28), I also updated the document to reflect the current setup.

The `iree-requirements.txt` file was being used to pin versions of the `iree-compiler` and `iree-runtime` that would be released together with this `iree-turbine` package. That was overlooked when https://github.com/iree-org/iree-turbine/pull/115 removed pins from that file. I've reworked the script to reference `iree-requirements-ci.txt` for now, but we can continue to iterate on the cross-project release process going forward (with docs this time).

I've updated the version in code from 2.3.0 to 2.5.0. We can push a release to https://github.com/iree-org/iree-turbine/releases and https://pypi.org/project/iree-turbine/#history using the new version next. Note that 2.3.1, 2.4.0, and 2.4.1 were released to PyPI without updating the code or GitHub releases.

### CODE REVIEW

**Comment by ScottTodd** [comment]
Good news: this fixed the https://github.com/iree-org/iree-turbine/actions/workflows/test_build_release.yml workflow that has been stalling. I'd like to investigate _why_ it fixed it, but I might just settle for it working for now...

**Comment by ScottTodd** [comment]
Tested locally. The instructions in the doc are accurate. The only hiccup I see is that `build_release.py` overwrites the `version_info.json` file instead of using a `.gitignore`'d file. Minor inconvenience :P

File: build_tools/build_release.py [hunk 1]
```diff
@@ -55,13 +63,13 @@ def write_version_info(args):
...
-    if args.core_post_version:
-        core_version += f".{eval_version(args.core_post_version)}"
-    info_dict["core-version"] = core_version
+    # Compute package-version.
+    package_version = eval_version(args.package_version)
+    if args.package_pre_version:
+        package_version += eval_version(args.package_pre_version)
+    if args.package_post_version:
+        package_version += f".{eval_version(args.package_post_version)}"
+    info_dict["package-version"] = package_version
```
Comment (marbre) [inline build_tools/build_release.py:72]: Yet another way to compute / parse the version numbers. I assume we want to harmonize this over the projects when switching iree-turbine to the new versioning scheme? Nothing for this PR, just considerations.

**Review by marbre — COMMENTED** [review]
Looks good to me but I am not able to an in depth review before tomorrow. But feel free to merge as the CI passes :partying_face:

**Comment by ScottTodd** [comment]
After merging this, I plan on running the build script locally and pushing 2.5.0 to PyPI and a GitHub release.

We could also have `setup.py` point to a specific version [range] of the iree-compiler and iree-runtime packages. Right now it gets versions from only `requirements.txt`, where the IREE packages are not mentioned at all.

**Comment by ScottTodd** [comment]
> The `docs/releasing.md` file was first introduced next to the `build_release.py` script in [nod-ai/SHARK-ModelDev#605](https://github.com/nod-ai/SHARK-ModelDev/pull/605). I don't see that file in this repository (iree-org/iree-turbine) though, despite [nod-ai/SHARK-ModelDev#655](https://github.com/nod-ai/SHARK-ModelDev/pull/655) claiming that it had been upstreamed. Since the packages involved have changed over the months (shark-turbine --> iree-turbine naming, see also #28), I also updated the document to reflect the current setup.

Ah, it was moved to https://github.com/iree-org/iree/blob/main/RELEASING.md#build-promotion. I'd like to merge that file with https://iree.dev/developers/general/release-management/ to have a single place to look.

**Pull request merged**